contracts/StandardBounty.sol

```diff
@@ -0,0 +1,319 @@
+  {
+      hasPaidOut = true;
+      Fulfillment storage fulfillment = fulfillments[_fulfillmentId];
+      for (uint256 i = 0; i < _payoutTokens.length; i++){
+        for (uint256 j = 0; j < fulfillment.fulfillers.length; j++){
+          if (_payoutTokens[i] == address(0)){
+            require(this.balance >= _tokenAmounts[i]);
+            fulfillment.fulfillers[j].transfer(calculateFraction(_tokenAmounts[i], fulfillment.numerators[j], fulfillment.denomenator));
+          } else {
+            require(_payoutTokens[i].balanceOf(this) >= _tokenAmounts[i]);
+            require(_payoutTokens[i].transfer(fulfillment.fulfillers[j], calculateFraction(_tokenAmounts[i], fulfillment.numerators[j], fulfillment.denomenator)));
+          }
+        }
+      }
+
+      FulfillmentAccepted(_fulfillmentId);
+  }
+
+  function acceptAndFulfill(address[] _fulfillers, uint[] _numerators, uint _denomenator, string _data, StandardToken[] _payoutTokens, uint[] _tokenAmounts)
+      public
+      sameLength(_payoutTokens.length, _tokenAmounts.length)
+      onlyController
+  {
+      fulfillBounty(_fulfillers, _numerators, _denomenator, _data);
+      hasPaidOut = true;
+      Fulfillment storage fulfillment = fulfillments[fulfillments.length - 1];
+      for (uint256 i = 0; i < _payoutTokens.length; i++){
+        for (uint256 j = 0; j < fulfillment.fulfillers.length; j++){
+          if (_payoutTokens[i] == address(0)){
+            require(this.balance >= _tokenAmounts[i]);
+            fulfillment.fulfillers[j].transfer(calculateFraction(_tokenAmounts[i], fulfillment.numerators[j], fulfillment.denomenator));
+          } else {
+            require(_payoutTokens[i].balanceOf(this) >= _tokenAmounts[i]);
+            require(_payoutTokens[i].transfer(fulfillment.fulfillers[j], calculateFraction(_tokenAmounts[i], fulfillment.numerators[j], fulfillment.denomenator)));
+          }
+        }
+      }
+
+      FulfillmentAccepted(fulfillments.length - 1);
+  }
+
+  function drainBounty(StandardToken[] _payoutTokens)
+      public
+      onlyController
+  {
+    for (uint256 i = 0; i < _payoutTokens.length; i++){
+      uint toPay;
+      if (_payoutTokens[i] == address(0x0)){
+        toPay = this.balance;
+        controller.transfer(toPay);
+
+      } else {
+        toPay = _payoutTokens[i].balanceOf(this);
+        require(_payoutTokens[i].transfer(controller, toPay));
+      }
+    }
+      BountyDrained(msg.sender);
+  }
+
+  function changeBounty(address _controller, string _data)
+      public
+      onlyController
+  {
+      controller = _controller;
+      data = _data;
+      BountyChanged();
+  }
+
+  function changeController(address _controller)
+      public
+      onlyController
+  {
+      controller = _controller;
+      BountyChanged();
+  }
+
+  function changeData(string _data)
+      public
+      onlyController
+  {
+      data = _data;
+      BountyChanged();
+  }
+
+  function changeMasterCopy(StandardBounty _masterCopy)
```

Comment:
hmmm. So we put this under the control of the bounty creator.  So if a bounty needs to be updated to a new master contract, the actual creator has to do that. Interesting.

Reply:
Would you prefer it to be us? That sounds awful and centralized!

Reply:
I assume there are serious security concerns if we lock that to the proxy creator haha.

Reply:
Definitely not. :D hahahahaha